### PR TITLE
Avoid an extra file db lookup

### PR DESCRIPTION
**Pull request by golenkovm**
This is a fix for #263

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -314,20 +324,34 @@ Openstack Object Storage settings
...
 
-| Moodle version   | Totara version     | Branch       | PHP  |Mandatory patches | Pre-signed URLs |
-|------------------|--------------------|--------------|------|------------------|-----------------|
-| Moodle 3.9       |                    | master       | 7.0+ |                  |                 |
-| Moodle 3.8       |                    | master       | 7.0+ | MDL-58281         | MDL-68342       |
-| Moodle 3.4 - 3.7 |                    | master       | 7.0+ | MDL-58281         | MDL-68342, MDL-66304 |
-| Moodle 3.3       | Totara 12          | master       | 7.0+ | MDL-58281         | MDL-68342, MDL-53240,<br>MDL-66304 |
-| Moodle 2.9 - 3.2 | Totara 2.9, 9 - 11 | 27-32-STABLE | 5.5+ | MDL-58281, MDL-46375,<br>MDL-58068, MDL-58684,<br>MDL-58297 | MDL-68342, MDL-53240,<br>MDL-66304 |
-| Moodle 2.7 - 2.8 | Totara 2.7 - 2.8   | 27-32-STABLE | 5.5+ | MDL-58281, MDL-46375,<br>MDL-58068, MDL-49627,<br>MDL-58684, MDL-58297 | MDL-68342, MDL-53240,<br>MDL-66304 |
+| Moodle version   | Totara version     | Branch       | Mandatory patches | Pre-signed URLs |
```
Comment (brendanheywood) [inline README.md:327]: why did you remove the php stuff from the readme?

Comment (golenkovm) [reply README.md:327]: I moved it to Branches section

File: classes/local/store/object_file_system.php [hunk 2]
```diff
@@ -401,6 +401,29 @@ public function get_content(\stored_file $file) {
...
+     *
+     * @param stored_file $file The file to send
+     * @return bool success
+     * @throws \dml_exception
+     */
+    public function xsendfile_file(stored_file $file): bool {
+        $contenthash = $file->get_contenthash();
+        if ($this->presigned_url_configured() &&
+                $this->presigned_url_should_redirect_file($file) &&
+                $this->is_file_readable_externally_by_hash($contenthash)) {
```
Comment (brendanheywood) [inline classes/local/store/object_file_system.php:420]: I would profile this, but I would expect a db call to be less overhead than an s3 call. It's probably very marginal.

But some moodles (not ours) will have objectfs remote, so I'd lean towards db first
